logic/dungeon3.py

```diff
@@ -5,91 +5,188 @@
-        pre_boss = Location(dungeon=3).connect(towards_boss4, AND(r.enemy_requirements["PAIRODD"], FEATHER, PEGASUS_BOOTS))
-        Location("D3 BEFORE NIGHTMARE").add(DroppedKey(0x15B)).connect(pre_boss, r.enemy_requirements["KEESE"])
-
-        boss_room = Location("D3 Boss Room", dungeon=3).connect(pre_boss, NIGHTMARE_KEY3)
-        boss = Location(dungeon=3).add(HeartContainer(0x15A), Instrument(0x159)).connect(boss_room, r.boss_requirements[world_setup.boss_mapping[2]])
+            north_4way.connect(north_4way_owl1, STONE_BEAK3, back=False)
+            after_b_stairs.connect(after_b_stairs_owl2, STONE_BEAK3, back=False)
+            fenced_walkway.connect(fenced_walkway_owl3, STONE_BEAK3, back=False)
+            
+        # connections
+        # entrance
+        entrance.connect(entrance_chest1, PEGASUS_BOOTS, back=False)
+        entrance.connect(after_pot_door, POWER_BRACELET, back=None)
+        after_pot_door.connect(after_pot_door_chest2, AND(r.enemy_requirements["GEL"], r.enemy_requirements["MOBLIN_SWORD"], r.enemy_requirements["BOUNCING_BOMBITE"]), back=False)
+        after_pot_door.connect(west_hallway, PEGASUS_BOOTS)
+        #TODO: slime_room.connect((after_pot_door, swordstalfos_room, before_a_stairs), r.enemy_requirements["HIDING_ZOL"], back=None) #TODO: ENABLE after removing casual logic patch
+        slime_room.connect(slime_room_chest3, back=False)
+        west_hallway.connect(west_hallway_clear, r.enemy_requirements["GEL"], back=False)
+        west_hallway.connect(before_a_stairs, back=False)
+        before_a_stairs.connect(before_a_stairs_chest5, AND("D3_GEL_CLEAR", r.enemy_requirements["STALFOS_EVASIVE"]), back=False)
+        swordstalfos_room.connect(swordstalfos_room_chest4, "SWITCH3", back=None)
+        before_a_stairs.connect(before_a_stairs_chest6, "SWITCH3", back=False)
+        before_a_stairs.connect(after_a_stairs)
+        # 4 way
+        after_a_stairs.connect((north_4way, south_4way, west_4way), FOUND(KEY3, 8), back=False)
+        north_4way.connect(north_4way_drop3, AND(r.enemy_requirements["STALFOS_AGGRESSIVE"], r.enemy_requirements["MOBLIN"]), back=False)
+        north_4way.connect(north_4way_switch, r.hit_switch, back=False)
+        south_4way.connect(south_4way_drop1, AND(r.enemy_requirements["HIDING_ZOL"], r.enemy_requirements["MOBLIN"], OR(r.enemy_requirements["PAIRODD"], BOOMERANG)), back=False)
+        west_4way.connect(west_4way_drop2, AND(r.enemy_requirements["HIDING_ZOL"], OR(r.enemy_requirements["PAIRODD"], BOOMERANG)), back=False)
+        after_a_stairs.connect(before_b_stairs, FOUND(KEY3, 4))
+        # main area
+        before_b_stairs.connect(after_b_stairs)
+        after_b_stairs.connect(after_b_stairs_drop4, r.enemy_requirements["HIDING_ZOL"], back=False)
+        after_b_stairs.connect(two_pairodd_room, AND(r.enemy_requirements["ZOL"], r.enemy_requirements["GEL"]), back=False)

```

Comment:
It's weird to have `back=False` here while in line 104 you define the backwards path anyway as a multiconnect. I would suggest to have `back=None` here and remove the connection to `after_b_stairs` in line 104 to improve readability